dockerfiles/cli/scripts/openshift/replace_stacks.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+# Copyright (c) 2012-2017 Red Hat, Inc
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+
+echo "[CHE] This script is going to replace Che stacks for current Che instance"
+
+command -v oc >/dev/null 2>&1 || { echo >&2 "[CHE] [ERROR] Command line tool oc (https://docs.openshift.org/latest/cli_reference/get_started_cli.html) is required but it's not installed. Aborting."; exit 1; }
+command -v jq >/dev/null 2>&1 || { echo >&2 "[CHE] [ERROR] Command line tool jq (https://stedolan.github.io/jq) is required but it's not installed. Aborting."; exit 1; }
+
+if [ -z "${CHE_API_ENDPOINT+x}" ]; then
+    echo -n "[CHE] Variable \$CHE_API_ENDPOINT is not set. Trying to infer it..."
+    che_host=$(oc get route che -o jsonpath='{.spec.host}')
+    if [ -z "${che_host}" ]; then echo >&2 "[CHE] [ERROR] Failed to infer environment variable \$CHE_API_ENDPOINT. Aborting. Please set it and run ${0} script again."; exit 1; fi
+    CHE_API_ENDPOINT="http://${che_host}/api"
+    echo "done (${CHE_API_ENDPOINT})"
+fi
+
+DEFAULT_NEW_STACKS_URL="https://raw.githubusercontent.com/redhat-developer/rh-che/master/builds/fabric8-che/assembly/fabric8-stacks/src/main/resources/stacks.json"
```

Comment:
maybe we should think how to avoid direct links to github, maybe it will be better to have this on central same as  OS yaml

Reply:
Publishing stacks.json in maven central is a good idea. That could publish the mechanisms to publish and share custom stacks for users. 

By the way this script (`replace_stack.sh`) is just a temporary workaround to use the docker images that are "OpenShift compatible". The long term solution should be to make all default docker images "OpenShift compatible" (c.f. point n.2 of   https://github.com/eclipse/che/issues/6097). When we have that we won't need to replace stacks.json anymore.